server/routes/users.js

```diff
@@ -0,0 +1,123 @@
+const Router = require('express')
+const User = require('../models/User')
+const router = new Router()
+const bcrypt = require('bcryptjs')
+const jwt = require('jsonwebtoken')
+
+router.post('/api/users', createUser)
+router.post('/api/user/login', logUser)
+router.get('/api/user', authenticateToken, getUserData)
+router.post('/api/user/logout', logoutUser)
+
+const createToken = (id) => jwt.sign(id.toString(), process.env.ACCESS_TOKEN)
+const maxAge = 24 * 60 * 60 * 1000
+
+async function createUser(req, res) {
+  const { password, email, name } = req.body
+
+  try {
+    const salt = await bcrypt.genSalt()
+    const hashedPassword = await bcrypt.hash(password, salt)
+
+    const users = await User.find()
```

Comment:
@AStashevskaya Its not optimal to query all users in database, for example if you have 10M+ users, it will consume a lot of time and memory. You can use `findOne` mongoose method to search user with same email and `count` to check, should you set `isAdmin` to true or not